packages/main/src/plugin/configuration-registry.ts

```diff
@@ -275,7 +275,7 @@ export class ConfigurationRegistry implements IConfigurationRegistry {
     const cloneConfig = { ...this.configurationValues.get(CONFIGURATION_DEFAULT_SCOPE) };
     // for each key being already the default value, remove the entry
     Object.keys(cloneConfig)
-      .filter(key => cloneConfig[key] === this.configurationProperties[key]?.default)
+      .filter(key => JSON.stringify(cloneConfig[key]) === JSON.stringify(this.configurationProperties[key]?.default))
```

Comment:
would it work using https://nodejs.org/api/util.html#utilisdeepstrictequalval1-val2 ?